### ISSUE TITLE
2FA App Store - Application specific passwords and FASTLANE_SESSION working once in a while.

### ISSUE THREAD

**Issue by celosotomaior** [fastlane/fastlane#15608]
I'm setting up a pipeline with Azure, which works fine for the first time, but when I try to use it again it doesn't use the application specific password anymore.

**Here's what I have done:**

Reinstalled fastlane.
Created a new application specific password at App Store.
Created a new FASTLANE_SESSION running the command spaceauth -u user@email.com on terminal.
It works once in a while, usually after reinstalling Fastlane.

Apart from this, the pipeline works fine. 

**Here is my pipeline:**
```pool:
  vmImage: 'macos-latest'

steps:
- task: InstallAppleCertificate@2
  inputs:
    certSecureFile: 'iOSDistributionCertificate.p12'
    certPwd: 'xxxxx'
    keychain: 'temp'

- task: InstallAppleProvisioningProfile@1
  inputs:
    provisioningProfileLocation: 'secureFiles'
    provProfileSecureFile: '4e932719-f9bd-4d3e-b4e6-08b35260c632.mobileprovision'

- task: ios-bundle-version@1
  inputs:
    sourcePath: 'BookingApp/info.plist'
    versionCodeOption: 'buildid'
    versionCode: '$(Build.BuildId)'
    versionCodeOffset: '0'
    printFile: true

- task: Xcode@5
  displayName: 'Build the app using Xcode'
  inputs:
    actions: 'build'
    sdk: 'iphoneos12.4'
    configuration: 'Release'
    scheme: 'BookingApp'
    packageApp: true
    xcodeVersion: 10 # Options: 8, 9, 10, default, specifyPath
    signingOption: 'auto'
    useXcpretty: false # Makes it easier to diagnose build failures
    xcWorkspacePath: '**/*.xcodeproj/project.xcworkspace' 

```

**Here's the output :**

```[command]fastlane pilot upload -u *** -i /Users/runner/runners/2.158.0/work/1/s/output/$(SDK)/$(Configuration)/BookingApp.ipa -q XXXXXXXX -r XXXXXX PLC
[15:59:22]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
[15:59:23]: Sending anonymous analytics information
[15:59:23]: Learn more at https://docs.fastlane.tools/#metrics
[15:59:23]: No personal or sensitive data is sent.
[15:59:23]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[15:59:23]: Login to App Store Connect (***)
Session loaded from environment variable is not valid. Continuing with normal login.
Two-factor Authentication (6 digits code) is enabled for account '***'
More information about Two-factor Authentication: https://support.apple.com/en-us/HT204915
If you're running this in a non-interactive session (e.g. server or CI)
check out https://github.com/fastlane/fastlane/tree/master/spaceship#2-step-verification
(Input `sms` to escape this prompt and select a trusted phone number to send the code as a text message)
(You can also set the environment variable `SPACESHIP_2FA_SMS_DEFAULT_PHONE_NUMBER` to automate this)
(Read more at: https://github.com/fastlane/fastlane/blob/master/spaceship/docs/Authentication.md#auto-select-sms-via-spaceship-2fa-sms-default-phone-number) 
```

Any idea on what could be the issue ? I'm using fastlane 2.28.3

Thanks in advance.

**Comment by Katsz** [comment]
Seems like it duplicates #15581.
Still encourage you not to use 2FA in CI/CD environment. Just create another apple id with 2FA disabled. 

 
 

Uh, yeah, and by the way.
fastlane 2.28.3 was released **two years ago**.

It's time for an upgrade
![image](https://user-images.githubusercontent.com/16171550/68433568-e6c4d380-01c7-11ea-8270-4341308b0e20.png)

**Comment by celosotomaior** [comment]
Thanks for the reply. 

The 2FA was already set on the account, and since I can't disable on this account, I will create a new Apple ID.

Yes, I installed through homebrew, because the command below described on documentation didn't work on my machine:

sudo gem install fastlane -NV 

Here's the output:
```
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /usr/bin directory.
```

After some research, I managed to install through the command: 
sudo gem install fastlane -NV -n /usr/local/bin

So now I have the latest version.

I also had to run the command 
sudo gem install bundler:1.17.3 -n /usr/local/bin
